src/NLog/Internal/TargetWithFilterChain.cs

```diff
@@ -92,17 +94,21 @@ internal void PrecalculateStackTraceUsage()
 
             // find all objects which may need stack trace
             // and determine maximum
-            foreach (var item in ObjectGraphScanner.FindReachableObjects<IUsesStackTrace>(this))
+            // only the target can have IUsesStackTrace
+            if (Target != null)
             {
-                var stu = item.StackTraceUsage;
-
-                if (stu > this.stackTraceUsage)
+                foreach (var item in Target.GetAllLayouts().OfType<IUsesStackTrace>())
```

Comment:
Afaik only LayoutRenderers implement IUsesStackTrace, not Layout, so OfType will yield nothing.